tasks/db_deploy/test/unit_tests/test_migrate_db.py

```diff
@@ -36,25 +36,110 @@ def tearDown(self):
 
                 # Make sure the proper migrations happens.
                 # Note that for version 2 and 3 the function is not called so
                 # overall through all the tests it is only called once.
+
                 if version == 1:
-                    mock_migrate_v1_to_v2.assert_called_once_with(self.config, True)
+                    mock_migrate_v1_to_v2.assert_called_once_with(self.config, False)
+                    mock_migrate_v2_to_v3.assert_called_once_with(self.config, True)
+                elif version == 2:
+                    mock_migrate_v2_to_v3.assert_called_once_with(self.config, True)
 
-                if version >= 2:
+                else:
                     mock_migrate_v1_to_v2.assert_not_called()
+                    mock_migrate_v2_to_v3.assert_not_called()
 
                 # Reset for next loop
                 mock_migrate_v1_to_v2.reset_mock()
+                mock_migrate_v2_to_v3.reset_mock()
+
+    @patch("migrate_db.schema_versions_data_sql")
+    @patch("migrate_db.providers_table_sql")
+    @patch("migrate_db.collections_table_sql")
+    @patch("migrate_db.provider_collection_xref_table_sql")
+    @patch("migrate_db.granules_table_sql")
+    @patch("migrate_db.files_table_sql")
+    @patch("migrate_db.get_admin_connection")
+    def test_migrate_versions_2_to_3_happy_path(
+        self,
+        mock_connection: MagicMock,
+        mock_files_table: MagicMock,
+        mock_granules_table: MagicMock,
+        mock_provider_collection_xref_table: MagicMock,
+        mock_collections_table: MagicMock,
+        mock_providers_table: MagicMock,
+        mock_schema_versions_data: MagicMock,
+    ):
+        """
+        Tests the migrate_versions_2_to_3 function happy path
+        """
+        for latest_version in [True, True]:
+            with self.subTest(latest_version=latest_version):
+                # Setup the mock object that conn.execute is a part of in
+                # the connection with block
+                mock_conn_enter = mock_connection().connect().__enter__()
+
+                # Run the function
+                migrate_db.migrate_versions_2_to_3(self.config, latest_version)
+
+                # Check that all of the functions were called the correct
+                # number of times with the proper values
+                mock_connection.assert_any_call(
+                    self.config, self.config["user_database"]
+                )
+
+                mock_providers_table.assert_called_once()
+                mock_collections_table.assert_called_once()
+                mock_provider_collection_xref_table.assert_called_once()
+                mock_granules_table.assert_called_once()
+                mock_files_table.assert_called_once()
+
+                # Validate logic switch and set the execution order
+                if latest_version:
+                    mock_schema_versions_data.assert_called_once()
+                    execution_order = [
```

Comment:
You will likely need to check that role and search path sql was run. See the v1_v2 migration test.